actor/v2action/service_instance_test.go

```diff
@@ -306,4 +306,251 @@ var _ = Describe("Service Instance Actions", func() {
+						BeforeEach(func() {
+							sharedToSpaceName = "shared-to-space-name"
+						})
+
+						It("returns the space guid and all warnings", func() {
+							Expect(executeErr).ToNot(HaveOccurred())
+							Expect(spaceGUID).To(Equal("shared-to-space-guid"))
+							Expect(warnings).To(Equal(Warnings{
+								"get-space-service-instances-warning",
+								"get-service-instance-shared-tos-warning"}))
+						})
+					})
+
+					Context("and is not shared with the specified space", func() {
+						BeforeEach(func() {
+							sharedToSpaceName = "some-other-space-name"
+						})
+
+						It("returns an error and all warnings", func() {
+							Expect(executeErr).To(MatchError(actionerror.ServiceInstanceNotSharedToSpaceError{ServiceInstanceName: serviceInstanceName}))
+							Expect(warnings).To(Equal(Warnings{
+								"get-space-service-instances-warning",
+								"get-service-instance-shared-tos-warning"}))
+						})
+					})
+				})
+
+				Context("and is not shared with the specified org", func() {
+					BeforeEach(func() {
+						sharedToOrgName = "some-other-org-name"
+					})
+
+					Context("and is shared with a space with the same name as the specified space", func() {
+						BeforeEach(func() {
+							sharedToSpaceName = "shared-to-space-name"
+						})
+
+						It("returns an error and all warnings", func() {
+							Expect(executeErr).To(MatchError(actionerror.ServiceInstanceNotSharedToSpaceError{ServiceInstanceName: serviceInstanceName}))
+							Expect(warnings).To(Equal(Warnings{
+								"get-space-service-instances-warning",
+								"get-service-instance-shared-tos-warning"}))
+						})
+					})
+				})
+			})
+
+			Context("when the service instance is not shared", func() {
+				BeforeEach(func() {
+					fakeCloudControllerClient.GetServiceInstanceSharedTosReturns(
+						[]ccv2.ServiceInstanceSharedTo{},
+						ccv2.Warnings{"get-service-instance-shared-tos-warning"},
+						nil,
+					)
+				})
+
+				It("returns an error and all warnings", func() {
+					Expect(executeErr).To(MatchError(actionerror.ServiceInstanceNotSharedToSpaceError{ServiceInstanceName: serviceInstanceName}))
+					Expect(warnings).To(Equal(Warnings{
+						"get-space-service-instances-warning",
+						"get-service-instance-shared-tos-warning"}))
+				})
+			})
+
+			Context("when getting the shared-to information fails", func() {
+				BeforeEach(func() {
+					fakeCloudControllerClient.GetServiceInstanceSharedTosReturns(
+						[]ccv2.ServiceInstanceSharedTo{},
+						ccv2.Warnings{"get-service-instance-shared-tos-warning"},
+						errors.New("some-shared-to-api-failure"),
+					)
+				})
+
+				It("returns an error and warnings", func() {
+					Expect(executeErr).To(MatchError("some-shared-to-api-failure"))
+					Expect(warnings).To(Equal(Warnings{
+						"get-space-service-instances-warning",
+						"get-service-instance-shared-tos-warning"}))
+				})
+			})
+		})
+
+		Context("when the service instance does not exist", func() {
```

Comment:
Please add a unit test for the general failure test case when looking up the service instance fails.

Reply:
Tahnks, new test added for the error case.